packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -235,17 +188,57 @@ export const BlocksDropdown = () => {
     return block.isInBlocksSelector ? [...currentKeys, key] : currentKeys;
   }, []);
 
-  const [blockSelected, setBlockSelected] = React.useState(Object.keys(blocks)[0]);
+  const [selectedOption, setSelectedOption] = React.useState(Object.keys(blocks)[0]);
 
   /**
-   * @param {string} optionKey - key of the heading selected
+   * Function that detect if the the block selected is an Image or a Code block.
    */
-  const selectOption = (optionKey) => {
-    toggleBlock(editor, blocks[optionKey].value);
+  const isLastBlockImageOrCode = () => {
+    const { selection } = editor;
 
-    setBlockSelected(optionKey);
+    if (!selection) return false;
+
+    const [currentImageORCodeBlock] = Editor.nodes(editor, {
+      at: selection,
+      match: (n) => n.type === 'image' || n.type === 'code',
+    });
 
-    if (isLastBlockImageOrCode(editor)) {
+    if (currentImageORCodeBlock) {
+      const [, currentNodePath] = currentImageORCodeBlock;
+
+      const isNodeAfter = Boolean(Editor.after(editor, currentNodePath));
+
+      return !isNodeAfter;
+    }
+
+    return false;
+  };
+
+  /**
+   * Manages the selection of a block and change the node inside the Slate Editor.
+   * @param {object} value - the value of the selected block option
```

Comment:
I don't understand what a block option is, can we either make this clearer or partially type it better?